src/eventsource.js

```diff
@@ -90,8 +91,15 @@ $.EventSource.prototype = {
      * @param {OpenSeadragon.EventHandler} handler - Function to call when event is triggered.
      * @param {Object} [userData=null] - Arbitrary object to be passed unchanged to the handler.
      * @param {Number} [priority=0] - Handler priority. By default, all priorities are 0. Higher number = priority.
+     * @returns {Boolean} - True if the handler was added, false if it was rejected
      */
     addHandler: function ( eventName, handler, userData, priority ) {
+
+        if(Object.prototype.hasOwnProperty.call(this.rejectedEventList, eventName)){
+            $.console.error(`Error adding handler for ${eventName}. ${this.rejectedEventList[eventName]}`);
+            return false;
```

Comment:
We don't actually need to return here... It would be fine if we added the event. The important thing is that we log the error.

Reply:
Not returning means that if for some reason we do fire that event it does go through.

On the other hand, returning false is nice because it allows the `addHandler` caller to know whether it worked, which is probably important. So yeah, probably good to leave it the way you have it. Just thinking things through here.

Reply:
I like the idea that the return value indicates whether the handler was added or not - it gives callers a way to know if their handler was rejected programmatically, not just by logging an error. Also, I think if we say we aren't adding the handler, it would be unexpected and confusing for it to get called.

Speaking of, I expanded on this concept to log an error at the time of raising an event too, so that it gives a hint that something is wrong on the firing side too. (I.e. if you've said an event can't be listened for, trying to fire it is a mistake.)